tests/sub_port_interfaces/sub_ports_helpers.py

```diff
@@ -292,6 +304,71 @@ def generate_and_verify_decap_traffic(duthost, ptfadapter, src_port, dst_port, i
     pytest_assert(pkt_in_buffer is True, "Expected packet not available:\n{}".format(pkt_in_buffer))
 
 
+def generate_and_verify_balancing_traffic(duthost, ptfhost, ptfadapter, src_port, dst_port, ip_src, ip_dst, pktlen, ttl):
+    """
+    Send TCP packets and verify balancing range between sub-ports.
+    Args:
+        duthost: DUT host object
+        ptfhost: PTF host object
+        ptfadapter: PTF adapter
+        src_port: Port of PTF
+        dst_port: Port of DUT
+        ip_src: Source IP address of PTF
+        ip_dst: Destination IP address of DUT
+        pkt_action: Packet action (forwarded or drop)
+        pktlen: packet length
+        ttl: Time to live
+    """
+    router_mac = duthost.facts['router_mac']
+    src_port_number = int(get_port_number(src_port))
+    src_mac = ptfadapter.dataplane.get_mac(0, src_port_number)
+    ip_src = ipaddress.ip_address(u'10.0.0.1')
```

Comment:
in test step 4, the packets have different source ip address, but here `ip_src` is fixed?

Reply:
The source IP address change in [generate_and_verify_balancing_traffic](https://github.com/OleksandrKozodoi/sonic-mgmt/blob/add_balancing_sub_ports_test_case/tests/sub_port_interfaces/sub_ports_helpers.py#L342) function.